org.adempiere.base/src/org/compiere/model/MUserQuery.java

```diff
@@ -300,7 +300,7 @@ public MUserQuery(Properties ctx, ResultSet rs, String trxName)
 	protected boolean beforeSave(boolean newRecord) {
 		if (getAD_Tab_ID() > 0) {
 			if (newRecord || is_ValueChanged(COLUMNNAME_AD_Tab_ID)) {
-				MTab tab = new MTab(getCtx(), getAD_Tab_ID(), get_TrxName());
+				MTab tab = MTab.get(getAD_Tab_ID());

```

Comment:
Hi @hengsin - I have not tested, but usually reading cache in beforeSave or afterSave always bring problems for 2Pack.

In this case the test would be a 2Pack containing the tab and then the user query, when that happens, the tab is not discovered because the MTab.get is read out of the 2pack transaction.